civisocial.php

```diff
@@ -0,0 +1,176 @@
+ */
+function civisocial_civicrm_enable() {
+  _civisocial_civix_civicrm_enable();
+}
+
+/**
+ * Implements hook_civicrm_disable().
+ *
+ * @link http://wiki.civicrm.org/confluence/display/CRMDOC/hook_civicrm_disable
+ */
+function civisocial_civicrm_disable() {
+  _civisocial_civix_civicrm_disable();
+}
+
+/**
+ * Implements hook_civicrm_upgrade().
+ *
+ * @link http://wiki.civicrm.org/confluence/display/CRMDOC/hook_civicrm_upgrade
+ */
+function civisocial_civicrm_upgrade($op, CRM_Queue_Queue $queue = NULL) {
+  return _civisocial_civix_civicrm_upgrade($op, $queue);
+}
+
+/**
+ * Implements hook_civicrm_managed().
+ *
+ * Generate a list of entities to create/deactivate/delete when this module
+ * is installed, disabled, uninstalled.
+ *
+ * @link http://wiki.civicrm.org/confluence/display/CRMDOC/hook_civicrm_managed
+ */
+function civisocial_civicrm_managed(&$entities) {
+  _civisocial_civix_civicrm_managed($entities);
+}
+
+/**
+ * Implements hook_civicrm_caseTypes().
+ *
+ * Generate a list of case-types
+ *
+ * Note: This hook only runs in CiviCRM 4.4+.
+ *
+ * @link http://wiki.civicrm.org/confluence/display/CRMDOC/hook_civicrm_caseTypes
+ */
+function civisocial_civicrm_caseTypes(&$caseTypes) {
+  _civisocial_civix_civicrm_caseTypes($caseTypes);
+}
+
+/**
+ * Implements hook_civicrm_angularModules().
+ *
+ * Generate a list of Angular modules.
+ *
+ * Note: This hook only runs in CiviCRM 4.5+. It may
+ * use features only available in v4.6+.
+ *
+ * @link http://wiki.civicrm.org/confluence/display/CRMDOC/hook_civicrm_caseTypes
+ */
+function civisocial_civicrm_angularModules(&$angularModules) {
+  _civisocial_civix_civicrm_angularModules($angularModules);
+}
+
+/**
+ * Implements hook_civicrm_alterSettingsFolders().
+ *
+ * @link http://wiki.civicrm.org/confluence/display/CRMDOC/hook_civicrm_alterSettingsFolders
+ */
+function civisocial_civicrm_alterSettingsFolders(&$metaDataFolders = NULL) {
+  _civisocial_civix_civicrm_alterSettingsFolders($metaDataFolders);
+}
+
+/**
+ * Include the settings page in civicrm navigation menu.
+ */
+function civisocial_civicrm_navigationMenu(&$params) {
+  $maxID = CRM_Core_DAO::singleValueQuery("SELECT max(id) FROM civicrm_navigation") + 300;
+
+  $civisocial_settings_url = "civicrm/admin/setting/preferences/civisocial";
+
+  $administerMenuID = CRM_Core_DAO::getFieldValue('CRM_Core_BAO_Navigation', 'Administer', 'id', 'name');
+
+  $params[$administerMenuID]['child'][$maxID + 1] = array(
+    'attributes' => array(
+      'label' => 'Civisocial Credentials',
+      'name' => 'Civisocial Credentials',
+      'url' => $civisocial_settings_url,
+      'permission' => 'administer CiviReport',
+      'operator' => NULL,
+      'separator' => NULL,
+      'parentID' => $administerMenuID,
+      'navID' => $maxID + 1,
+      'active' => 1,
+    ),
+  );
+}
+
+function civisocial_civicrm_buildForm($formName, &$form) {
+  if (!user_is_logged_in()) {
```

Comment:
user_is_logged_in() is a drupal function, we need to make code compatible for all CMS. You can use $userID = CRM_Core_Session::singleton()->get("userID"); to check if user is logged in or not